tests/utils/memory_tests.cpp

```diff
@@ -32,6 +32,142 @@
 #include "utils/memory_pool.hpp"
 #include "utils/timer_utils.hpp"
 
+namespace irs::memory {
+
+///////////////////////////////////////////////////////////////////////////////
+/// @class memory_multi_size_pool
+/// @brief an allocator that can handle allocations of slots with different
+///        sizes
+///////////////////////////////////////////////////////////////////////////////
+template<typename GrowPolicy = log2_grow,
+         typename BlockAllocator = malloc_free_allocator>
+class memory_multi_size_pool : public pool_base<GrowPolicy, BlockAllocator> {
```

Comment:
If this is now test only code - why not remove/modify tests that use it?

Reply:
I think it can be done separately.
Also this test code could not directly but test base classes, which is still used

Reply:
About memory pool I think it can be removed at all, I just needs benchmark